app/controllers/api/v1/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+        respond_to :json
+        def google_oauth2
+          @user = User.from_omniauth(auth)
+
+          # TODO: Return JWT token on both success and error
```

Comment:
JWT Tokens and OAuth tokens are not the same thing ⚠️ So I recommend to to do more research to understand the differences to deepen your understanding of API. Here is quick insight: https://stackoverflow.com/a/39914013